### PR TITLE
Simplify error checking & reporting

### PR DESCRIPTION
**Pull request by aerosol**
Hi,

Ref https://github.com/parroty/excheck/issues/36#issuecomment-312560270

Capturing IO is no longer necessary.
Counterexamples are always printed with regular ExUnit errors.
Property test iterations and failures are formatted to distinguish them from regular ExUnit tests.

image

image

@parroty please let me know what you think, I could push this further

### CODE REVIEW

File: lib/excheck.ex [hunk 1]
```diff
@@ -40,19 +39,27 @@ defmodule ExCheck do
...
-        false
-      {:EXIT, %{__struct__: ExUnit.AssertionError} = error} ->
-        raise ExUnit.AssertionError, message: error.message
-      {:EXIT, %{__struct__: type, message: msg}} ->
-        raise ExCheck.Error, message: "error raised: (#{type}) #{msg}"
+        example = :triq.counterexample()
+        {:error, %ExCheck.Error{message: "check failed: Counterexample: #{inspect example}"}}
+      {:EXIT, %{__struct__: _, message: _} = e} ->
+        example = :triq.counterexample()
+        {:error, e}
```
Comment (aerosol) [inline lib/excheck.ex]: fixme. no counter example here

**Comment by aerosol** [comment]
The PR is still work in progress, I'm looking into providing better ExUnit integration.

**Comment by parroty** [comment]
@aerosol Thanks for the updates. Just confirmation, but the above ExUnit integration part is in progress?

**Comment by aerosol** [comment]
@parroty Hey, sorry this is taking so long.

Yes, I am still working on making it better and I am not _too happy_ with the state of this PR. 
My initial idea evolved over time, but I haven't had enough mental capacity to pull it off quickly.

If you feel you like it, go ahead and merge and I will submit more fixes on top of it. 
It does solve _some_ problems with `excheck` after all.

One way or another, I need more time to properly implement what I have in mind and that will eventually land in this or another PR. Thanks for keeping in touch! <3

**Pull request closed without merging**